### PR TITLE
perf: remove bounds check in DUP, SWAP/EXCHANGE

### PR DESCRIPTION
**Pull request by DaniPopes**
It was introduced by making `n` dynamic instead of constant

### CODE REVIEW

File: crates/interpreter/src/interpreter/stack.rs [hunk 1]
```diff
@@ -222,8 +222,14 @@ impl Stack {
...
 
     /// Duplicates the `N`th value from the top of the stack.
+    ///
+    /// # Panics
+    ///
+    /// Panics if `n` is 0.
     #[inline]
+    #[cfg_attr(debug_assertions, track_caller)]
     pub fn dup(&mut self, n: usize) -> Result<(), InstructionResult> {
+        assume!(n > 0, "attempted to dup 0");
```
Comment (rakita) [inline crates/interpreter/src/interpreter/stack.rs:232]: What does `assume` do here, if we are checking `n >0`, as N is usize, it always will be more than 0?

Comment (DaniPopes) [reply crates/interpreter/src/interpreter/stack.rs:232]: n == 0 doesn't make sense, there is no dup0 or swap0

**Review by onbjerg — APPROVED** [review]

File: crates/interpreter/src/interpreter_action/call_inputs.rs [hunk 2]
```diff
@@ -36,7 +36,7 @@ pub struct CallInputs {
     ///
     /// Previously `context.scheme`.
     pub scheme: CallScheme,
-    /// Whether the call is initiated inside a static call.
+    /// Whether the call is, or is initiated inside a static call.
```
Comment (onbjerg) [inline crates/interpreter/src/interpreter_action/call_inputs.rs:39]: think a word is missing here

File: crates/interpreter/src/interpreter/stack.rs [hunk 3]
```diff
@@ -232,36 +238,48 @@ impl Stack {
...
         let n_m_index = n + m;
         if n_m_index >= len {
             return Err(InstructionResult::StackUnderflow);
         }
-        let last_index = len - 1;
-        self.data.swap(last_index - n, last_index - n_m_index);
+        // SAFETY: `n` and `n_m` are checked to be within bounds, and they don't overlap.
+        unsafe {
+            // NOTE: `mem::swap` performs better than `slice::swap`/`ptr::swap`, as they cannot
+            // assume that the pointers are non-overlapping when we know they are not.
```
Comment (onbjerg) [inline crates/interpreter/src/interpreter/stack.rs:279]: just to clarify: when we know they are not overlapping, right? this reads as "we know they are not non-overlapping" → "we know they are overlapping"

File: crates/interpreter/src/interpreter_action/call_inputs.rs (hunk 2 above)
Comment (DaniPopes) [reply crates/interpreter/src/interpreter_action/call_inputs.rs:39]: is,
or
is initialized inside

I think this is correct

File: crates/interpreter/src/interpreter/stack.rs (hunk 3 above)
Comment (DaniPopes) [reply crates/interpreter/src/interpreter/stack.rs]: Yeah it's not great, chat gpt'd it

File: crates/interpreter/src/interpreter_action/call_inputs.rs (hunk 2 above)
Comment (DaniPopes) [reply crates/interpreter/src/interpreter_action/call_inputs.rs]: rewrote it anyway

Comment (onbjerg) [reply crates/interpreter/src/interpreter_action/call_inputs.rs]: ah i see what this means now, yeah i think it's correct but it's hard to parse 😅

File: Cargo.toml [hunk 4]
```diff
@@ -14,11 +14,6 @@ default-members = ["crates/revm"]
 all-features = true
 rustdoc-args = ["--cfg", "docsrs"]
 
-[profile.release]
-lto = true
-codegen-units = 1
```
Comment (rakita) [inline Cargo.toml:19]: I would leave `lto` and `codegen-units` as they can help with performance.

Comment (DaniPopes) [reply Cargo.toml]: remnants from debugging
